soundata/datasets/dcase23_task4b.py

```diff
@@ -0,0 +1,276 @@
+Creative Commons Attribution 4.0 International
+"""
+
+
+class Clip(core.Clip):
+    """DCASE23_Task4B Clip class
+
+    Args:
+        clip_id (str): id of the clip
+
+    Attributes:
+        audio (np.ndarray, float): path to the audio file
+        audio_path (str): path to the audio file
+        annotations_path (str): path to the annotations file
+        clip_id (str): clip id
+        events (soundata.annotations.Events): sound events with start time,
+            end time, label and confidence
+        split (str): subset the clip belongs to:
+            development or evaluation
+    """
+
+    def __init__(self, clip_id, data_home, dataset_name, index, metadata):
+        super().__init__(clip_id, data_home, dataset_name, index, metadata)
+
+        self.audio_path = self.get_path("audio")
+        self.annotations_path = self.get_path("annotations")
+
+    @property
+    def audio(self) -> Optional[Tuple[np.ndarray, float]]:
+        """The clip's audio
+
+        Returns:
+            * np.ndarray - audio signal
+            * float - sample rate
+
+        """
+        return load_audio(self.audio_path)
+
+    @property
+    def split(self):
+        """The clip's split.
+
+        Returns:
+            * str - subset the clip belongs to: development or evaluation
+
+        """
+        return self._clip_metadata.get("split")
+
+    @core.cached_property
+    def events(self) -> Optional[annotations.Events]:
+        """The clip's events.
+
+        Returns:
+            * annotations.Events - sound events with start time, end time, label and confidence
+
+        """
+        return load_events(self.annotations_path)
+
+    def to_jams(self):
+        """Get the clip's data in jams format
+
+        Returns:
+            jams.JAMS: the clip's data in jams format
+
+        """
+        return jams_utils.jams_converter(
+            audio_path=self.audio_path, events=self.events, metadata=self._clip_metadata
+        )
+
+
+@io.coerce_to_bytes_io
+def load_audio(fhandle: BinaryIO, sr=None) -> Tuple[np.ndarray, float]:
+    """Load a DCASE23_Task4B audio file.
+
+    Args:
+        fhandle (str or file-like): File-like object or path to audio file
+        sr (int or None): sample rate for loaded audio, None by default, which
+            uses the file's original sample rate of 44100 without resampling.
+
+    Returns:
+        * np.ndarray - the stereo audio signal
+        * float - The sample rate of the audio file
+
+    """
+    audio, sr = librosa.load(fhandle, sr=sr, mono=False)
+    return audio, sr
+
+
+@io.coerce_to_string_io
+def load_events(fhandle: TextIO) -> annotations.Events:
+    """Load a DCASE23_Task4B annotation file
+    Args:
+        fhandle (str or file-like): File-like object or path to the sound
+        events annotation file
+
+    Returns:
+        Events: sound events annotation data
+    """
+
+    times = []
+    labels = []
+    confidence = []
+    reader = csv.reader(fhandle, delimiter="\t")
```

Comment:
I personally prefer the use of [DictReader](https://docs.python.org/3/library/csv.html#csv.DictReader), since then it's more easy to follow what are the csv fields. I find it difficult to follow with `line[0]`, `line[1]`, etc. Pandas could be used too, but I think DictReader is more efficient.